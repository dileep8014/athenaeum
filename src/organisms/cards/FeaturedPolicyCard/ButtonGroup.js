import React from 'react';
import PropTypes from 'prop-types';

import Button from 'atoms/Button';
import Spacer from 'atoms/Spacer';

import styles from './featured_policy_card.module.scss';

function ButtonGroup(props) {
  const {
    onDetails,
    onCompare,
  } = props;

  return (
    <div className={styles['button-group']}>
      <Button
        onClick={onDetails}
      >
        View Policy
      </Button>
      {
        onCompare &&
        <div>
          <Spacer spacer={2} />
          <Button
            outline
            onClick={onCompare}
          >
            Compare
          </Button>
        </div>
      }
    </div>
  );
}

ButtonGroup.propTypes = {
  onDetails: PropTypes.func.isRequired,
  onCompare: PropTypes.func
};

export default ButtonGroup;
