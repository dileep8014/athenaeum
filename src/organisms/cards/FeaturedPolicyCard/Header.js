import React from 'react';
import PropTypes from 'prop-types';

import styles from './featured_policy_card.module.scss';

const Header = ({ header }) =>
  <div className={styles['header']}>
    {
      header.map((item, idx) =>
        <div
          key={`header-item-${idx}`}
          className={styles['header-item']}
        >
          {item}
        </div>
      )
    }
  </div>

;

Header.propTypes = {
  header: PropTypes.arrayOf(PropTypes.node),
};

export default Header;
