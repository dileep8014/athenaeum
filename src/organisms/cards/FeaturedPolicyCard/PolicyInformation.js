import React from 'react';
import PropTypes from 'prop-types';

import Col from 'atoms/Layout/Col';
import Layout from 'atoms/Layout';
import Spacer from 'atoms/Spacer';
import Text from 'atoms/Text';
import Tooltip from 'atoms/Tooltip';
import formatValue from 'utils/formatValue';

import styles from './featured_policy_card.module.scss';

const renderPolicyInformation = (info, idx) =>
  <div key={`policy-info-${idx}`}>
    <Layout
      smallCols={[ 4, 2, 6 ]}
      className={styles['policy-info-item']}
    >
      <Col
        fullwidth
        className={styles['policy-info-tooltip-group']}
      >
        <Text
          type={7}
          className={styles['policy-info-label']}
        >
          {info.label}
        </Text>
        <Tooltip
          className={styles['policy-info-tooltip-icon']}
        >
          {info.hoverMessage}
        </Tooltip>
      </Col>
      <div />
      <Text type={6} semibold>{formatValue(info.value)}</Text>
    </Layout>
    <Spacer spacer={2} />
  </div>

;

function PolicyInformation(props) {
  const {
    information
  } = props;

  return (
    <div className={styles['policy-info']}>
      {
        information.map(renderPolicyInformation)
      }
      <Spacer spacer={4} />
    </div>
  );
}

PolicyInformation.propTypes = {
  information: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    hoverMessage: PropTypes.node,
  })).isRequired,
};

export default PolicyInformation;
