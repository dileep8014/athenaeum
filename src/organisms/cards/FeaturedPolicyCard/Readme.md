### FeaturedPolicyCard Example:

```jsx
    <FeaturedPolicyCard
      information={[
        {
          label: 'Policy Type',
          value: 'Level Term',
          hoverMessage: <Text>This is information about the policy type.</Text>
        },
        {
          label: 'Financial Strength',
          value: 'A+',
          hoverMessage: <Text>This is information about the financial strength</Text>
        },
        {
          label: 'Total Customers',
          value: 16000000,
          hoverMessage: <Text>This is information about total customers.</Text>
        },
      ]}
      carrierLogo={
        <img
          src='http://res.cloudinary.com/policygenius/image/upload/v1500328467/mutual-of-omaha-dark.svg'
          alt='hello'
        />
      }
      premium={{
        price: 26.73,
        format: 'mo',
        defaultText: 'Quote available from a PolicyGenius expert'
      }}
      discount={
        <Text color='neutral-1' type={7}>
          Save 5% when paying annually
        </Text>
      }
      onContinue={() => alert('continue button clicked')}
      onDetails={() => alert('details button clicked')}
      onCompare={() => alert('compare button clicked')}
      header={[
        <Icon icon='checkmark' height='16px' width='16px' />,
        <span style={{ fontSize: '13px', color: '#78c07f', fontWeight: 'bold' }}>Great rates for peoeple with diabetes</span>,
      ]}
    />
```
