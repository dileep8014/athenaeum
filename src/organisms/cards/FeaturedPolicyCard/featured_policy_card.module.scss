$card-border: 1px solid #dfdfdf;
$radius: rem-calc(8px);
$details-link-color: #4a4a4a;
$details-link-border: 1px solid #b2b2b2;
$policy-hat-height: ru(.25);
$content-margin: ru(1);

.featured-policy-card {
  border: $card-border;
  background-color: color('neutral-8');
  display: flex;
  flex-direction: column;
  height: 100%;

  .details-link {
    @include typography-7(false);

    padding-bottom: 2px;
    color: $details-link-color;
    border-bottom: $details-link-border;
    text-decoration: none;
    display: none;

    &:hover {
      border-color: $details-link-color;
    }
  }
}

.content {
  margin: $content-margin;
  background-color: color('neutral-8');
  border-radius: $radius;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  flex-grow: 1;

  &.policy-hat {
    position: relative;

    &:before {
      position: absolute;
      content: '';
      display: block;
      height: $policy-hat-height;
      background: color('secondary-4');
      width: 100%;
      top: calc(-#{$policy-hat-height} + -#{$content-margin});
      left: 0;
    }
  }
}

.premium {
  text-align: center;
}

.carrier-logo {
  align-items: center;
  display: flex;
  height: rem-calc(60px);
  justify-content: center;
  max-width: rem-calc(170px);

  img {
    max-width: 100%;
    max-height: 100%;
  }
}

.policy-info {
  width: 100%;

  &-item {
    align-items: center;
  }

  &-tooltip-group {
    text-align: right;
    align-items: center;
  }

  &-tooltip-icon {
    margin-left: ru(.25);
    margin-right: auto;
  }
}

.button-group {
  width: 100%;
  text-align: center;
  margin-top: auto;
}

.secondary-action {
  &:first-child {
    padding-right: ru(.25);
  }

  &:last-child {
    padding-left: ru(.25);
  }
}

@media #{$medium-up} {
  .featured-policy-card {
    .details-link {
      display: inline-block;
    }
  }

  .secondary-actions-wrapper {
    display: none;
  }
}
