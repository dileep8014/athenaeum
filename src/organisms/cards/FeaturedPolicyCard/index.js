import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import accounting from 'accounting';

import Text from 'atoms/Text';
import Spacer from 'atoms/Spacer';

import styles from './featured_policy_card.module.scss';
import ButtonGroup from './ButtonGroup';
import PolicyInformation from './PolicyInformation';

function FeaturedPolicyCard(props) {
  const {
    carrierLogo,
    className,
    continueCTAText,
    discount,
    header,
    information,
    onContinue,
    onDetails,
    onCompare,
    policyHat,
    premium,
  } = props;

  const classes = [
    styles['featured-policy-card'],
    className,
  ];

  const contentClasses = classnames(
    styles['content'],
    policyHat && styles['policy-hat'],
  );

  const premiumValues = accounting.toFixed(premium.price, 2).split('.');
  const dollars = premiumValues[0];
  const cents = premiumValues[1];

  return (
    <div className={classnames(...classes)}>
      <div className={contentClasses}>
        { header && header }
        <Spacer small />
        { premium.price ?
          <div className={styles['premium']}>
            <div className={styles['value']}>
              <Text
                size={6}
                font='a'
                className={styles['value-small']}
              >
                $
              </Text>
              <Text
                size={2}
                font='a'
                className={styles['value-dollars']}
              >
                {dollars}
              </Text>
              <Text
                size={6}
                font='a'
                className={styles['value-small']}
              >
                .{cents}
              </Text>
              <Text
                tag='span'
                type={11}
                font='a'
                spaced
                color='neutral-2'
                className={styles['value-format']}
              >
                {`/${premium.format.toUpperCase()}`}
              </Text>
            </div>

            <Spacer spacer={1} />

            {discount && discount}
          </div>
            :
          <Text type={7} color='neutral-2'className={styles['default-text']}>{premium.defaultText}</Text>
          }

        <Spacer spacer={6} />

        <div className={styles['carrier-logo']}>
          { carrierLogo }
        </div>

        <Spacer spacer={6} />

        { information && <PolicyInformation information={information} /> }

        <ButtonGroup
          onContinue={onContinue}
          onDetails={onDetails}
          onCompare={onCompare}
          continueCTAText={continueCTAText}
        />
      </div>
    </div>
  );
}


FeaturedPolicyCard.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Array of nodes to display in header
   */
  header: PropTypes.arrayOf(PropTypes.node),

  /**
   * Displays premium value
   *
   * `defaultText` replaces `price` is `price` is not available
   */
  premium: PropTypes.shape({
    price: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    format: PropTypes.string,
    defaultText: PropTypes.string,
  }),

  /**
   * Displays discount message below premium
   */
  discount: PropTypes.node,

  /**
   * Image for carrier logo
   */
  carrierLogo: PropTypes.node,

  /**
   * Text displayed for the main action button
   */
  continueCTAText: PropTypes.string,

  /**
   * Function supplied to details link below CTA
   */
  onContinue: PropTypes.func.isRequired,

  /**
   * Function supplied to details link below CTA
   */
  onDetails: PropTypes.func,

  /**
   * Function supplied to compare CTA on mobile only
   */
  onCompare: PropTypes.func,

  /**
   * Supplies information about policy to card. Examples would include type, financial strength or total customers
   */
  information: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
    hoverMessage: PropTypes.node.isRequired,
  })),
  /**
   * Setting this to true will add a 'hat' to the policy card
   */
  policyHat: PropTypes.bool,
};

FeaturedPolicyCard.defaultProps = {
  continueCTAText: 'Continue',
  detailsCTAText: 'Details & Review',
  policyHat: false
};

export default FeaturedPolicyCard;
