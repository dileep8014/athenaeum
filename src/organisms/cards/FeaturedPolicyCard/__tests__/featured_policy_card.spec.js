import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import FeaturedPolicyCard from 'organisms/cards/FeaturedPolicyCard';

describe('<FeaturedPolicyCard />', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      header: [],
      carrierLogo: <div />,
      policyType: {
        value: 'A+',
        tooltipMessage: <div />
      },
      financialStrength: {
        value: 'A+',
        tooltipMessage: <div />
      },
      totalCustomers: {
        value: 16,
        tooltipMessage: <div />
      },
      premium: {
        price: 19.8,
        format: 'mo'
      },
      onContinue: jest.fn(),
      onDetails: jest.fn(),
      discount: <div />
    };
  });

  it('renders correctly', () => {
    const wrapper = mount(<FeaturedPolicyCard {...defaultProps} />);

    expect(wrapper.type()).toEqual(FeaturedPolicyCard);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<FeaturedPolicyCard {...defaultProps} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
