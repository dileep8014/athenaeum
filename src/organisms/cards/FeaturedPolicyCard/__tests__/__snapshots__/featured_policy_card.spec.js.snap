// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedPolicyCard /> renders correctly 1`] = `
<div
  className="featured-policy-card"
>
  <div
    className="content"
  >
    <div
      className="top-2 top-4"
    />
    <div
      className="premium"
    >
      <strong
        className="primary-3 type-a-3-bold inherit"
      >
        $19.80
        <strong
          className="primary-3 type-a-11-bold inherit"
        >
          /MO
        </strong>
      </strong>
      <div
        className="top-1"
      />
      <div />
    </div>
    <div
      className="top-6"
    />
    <div
      className="carrier-logo"
    >
      <div />
    </div>
    <div
      className="top-6"
    />
    <div
      className="policy-info"
    >
      <div>
        <div
          className="layout policy-info-item"
          id={undefined}
          style={undefined}
        >
          <div
            className="col col-sm-6 fullwidth border-neutral-6 policy-info-tooltip-group"
            onClick={undefined}
            style={undefined}
          >
            <span
              className="primary-3 type-b-10-medium inherit"
            >
              Financial Strength
            </span>
            <span
              className="tooltip-wrapper policy-info-tooltip-icon"
              onClick={[Function]}
            >
              <span
                className="icon-wrapper tooltip"
                onClick={undefined}
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="hover-message"
              >
                <div />
              </span>
            </span>
          </div>
          <div
            className="col col-sm-6 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <strong
              className="primary-3 type-a-8-bold inherit"
            >
              A+
            </strong>
          </div>
        </div>
        <div
          className="top-2"
        />
      </div>
      <div>
        <div
          className="layout policy-info-item"
          id={undefined}
          style={undefined}
        >
          <div
            className="col col-sm-6 fullwidth border-neutral-6 policy-info-tooltip-group"
            onClick={undefined}
            style={undefined}
          >
            <span
              className="primary-3 type-b-10-medium inherit"
            >
              Total Customers
            </span>
            <span
              className="tooltip-wrapper policy-info-tooltip-icon"
              onClick={[Function]}
            >
              <span
                className="icon-wrapper tooltip"
                onClick={undefined}
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="hover-message"
              >
                <div />
              </span>
            </span>
          </div>
          <div
            className="col col-sm-6 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <strong
              className="primary-3 type-a-8-bold inherit"
            >
              16,000,000
            </strong>
          </div>
        </div>
        <div
          className="top-2"
        />
      </div>
      <div>
        <div
          className="layout policy-info-item"
          id={undefined}
          style={undefined}
        >
          <div
            className="col col-sm-6 fullwidth border-neutral-6 policy-info-tooltip-group"
            onClick={undefined}
            style={undefined}
          >
            <span
              className="primary-3 type-b-10-medium inherit"
            >
              Policy Type
            </span>
            <span
              className="tooltip-wrapper policy-info-tooltip-icon"
              onClick={[Function]}
            >
              <span
                className="icon-wrapper tooltip"
                onClick={undefined}
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="hover-message"
              >
                <div />
              </span>
            </span>
          </div>
          <div
            className="col col-sm-6 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <strong
              className="primary-3 type-a-8-bold inherit"
            >
              Level Term
            </strong>
          </div>
        </div>
        <div
          className="top-2"
        />
      </div>
      <div
        className="top-4"
      />
    </div>
    <div
      className="button-group"
    >
      <button
        className="button"
        disabled={undefined}
        onClick={[Function]}
        type="button"
      >
        View Policy
      </button>
    </div>
  </div>
</div>
`;
